server-timing/load.php

```diff
@@ -113,3 +116,72 @@ function perflab_wrap_server_timing( $callback, $metric_slug, $access_cap ) {
 		return $result;
 	};
 }
+
+/**
+ * Registers the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_server_timing_setting() {
+	register_setting(
+		PERFLAB_SERVER_TIMING_SCREEN,
+		PERFLAB_SERVER_TIMING_SETTING,
+		array(
+			'type'              => 'object',
+			'sanitize_callback' => 'perflab_sanitize_server_timing_setting',
+			'default'           => array(),
+		)
+	);
+}
+add_action( 'init', 'perflab_register_server_timing_setting' );
+
+/**
+ * Sanitizes the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $value Server-Timing setting value.
+ * @return array Sanitized Server-Timing setting value.
+ */
+function perflab_sanitize_server_timing_setting( $value ) {
+	if ( ! is_array( $value ) ) {
+		return array();
+	}
+
+	/*
+	 * Ensure that every element is an indexed array of hook names.
+	 * Any duplicates across a group of hooks are removed.
+	 */
+	return array_filter(
+		array_map(
```

Comment:
Should this be checking whether the keys are those which are allowed, i.e. `benchmarking_actions` and `benchmarking_filters`?

Reply:
Good idea, updated in https://github.com/WordPress/performance/pull/784/commits/7f99021dd4afe5420dc126a96b6d26a8ea0bf16c